### DISCUSSION TITLE
Problem about compile a new primal solver

### DISCUSSION THREAD

**Discussion by DUT-wansy** [mdolab/dafoam#308] [General]
Hello Dr He,
For some reason I need to use SRFSimpleFoam as my solver, but I got this error during compilation, Could you please offer me some good suggestions?
Maybe this error is because there are no SRF related classes in DaFoam and I need to be redefined for compilation?
```shell
In file included from DASolver/DASRFSimpleFoam/DASRFSimpleFoam.C:158:
DASolver/DASRFSimpleFoam/UEqnSRFSimple.H: In member function ‘virtual Foam::label Foam::DASRFSimpleFoam::solvePrimal(Vec, Vec)’:
DASolver/DASRFSimpleFoam/UEqnSRFSimple.H:14:7: error: expected primary-expression before ‘->’ token
   14 |  + SRF->Su()
      |       ^~
DASolver/DASRFSimpleFoam/DASRFSimpleFoam.C:162:23: error: expected primary-expression before ‘->’ token
  162 |         Uall = U + SRF->U(); // 定义绝对速度

```
Thank you!

**Comment by friedenhe** [comment]
It is hard to tell from the code snippet. You can either fork the repo and push your update to your fork or share the new codes you added here.

**Reply by DUT-wansy** [reply]
OK, In my solver, I just changed `DASimpleFoam/UEqnSimple.H`.


The modified code :
```cpp
// Momentum predictor

// MRF.correctBoundaryVelocity(U);

if (hasFvSource_)
{
    daFvSourcePtr_->calcFvSource(fvSource);
}

tmp<fvVectorMatrix> tUEqn(
    fvm::div(phi, U)
    + fvm::Sp(alphaPorosity, U)
    // + MRF.DDt(U)
	+ SRF->Su()
    + turbulencePtr_->divDevReff(U)
    - fvSource
    - fvOptions(U));
fvVectorMatrix& UEqn = tUEqn.ref();

UEqn.relax();

fvOptions.constrain(UEqn);

if (simple.momentumPredictor())
{
    // get the solver performance info such as initial
    // and final residuals
    SolverPerformance<vector> solverU = solve(UEqn == -fvc::grad(p));

    this->primalResidualControl<vector>(solverU, printToScreen, printInterval, "U");
}

```

I have already  changed SimpleFoam to SRFSimpleFoam in OpenFoam in the same way and verified it with an case.

**Reply by friedenhe** [reply]
Where did you create the SRF object?

**Reply by DUT-wansy** [reply]
OK, Thank you, I'm trying to create the SRF object.
By the way, Is the process of creating the SRF object the same as that of creating the MRF object?

**Reply by DUT-wansy** [reply]
Or can I compile SRF model in OpenFOAM directly?

**Reply by friedenhe** [reply]
This is an OpenFOAM question. I suggest you post it to CFD-online.com
